### PR TITLE
The Draggable able change cursor on dragging

### LINKED ISSUE #92083: MouseRegion's cursor in a Draggable's feedback does not work
## Steps to Reproduce

Consider this minimal example ([dartpad](http://dartpad.dev/498f3e532296ab72a9cce857484b762f)):

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: Draggable(
            child: const MouseRegion(
              cursor: SystemMouseCursors.grab,
              child: Square(color: Colors.red),
            ),
            childWhenDragging: Square(color: Colors.red.shade100),
            feedback: const MouseRegion(
              cursor: SystemMouseCursors.grabbing,
              child: Square(color: Colors.red),
            ),
          ),
        ),
      ),
    );
  }
}

class Square extends StatelessWidget {
  const Square({
    required this.color,
    Key? key,
  }) : super(key: key);

  final Color color;

  @override
  Widget build(BuildContext context) {
    return ColoredBox(
      color: color,
      child: const SizedBox.square(dimension: 64.0),
    );
  }
}
```

**Expected results:**

While hovering the square area, the cursor should be `grab`. While dragging, the cursor should be `grabbing`.

**Actual results:**

While hovering the square area, the cursor is `grab`, as expected. When we click to start dragging, the cursor reverts to the "normal" cursor instead of going to `grabbing`.

<details>
  <summary>flutter doctor -v</summary>

```
➜  ~ flutter doctor -v
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-x64, locale
    en-BR)
    • Flutter version 2.5.3 at /Users/pintome/Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (4 days ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/pintome/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.0, Build version 13A233
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (3 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API
      30) (emulator)
    • macOS (desktop)         • macos         • darwin-x64     • macOS 11.6
      20G165 darwin-x64
    • Chrome (web)            • chrome        • web-javascript • Google Chrome
      94.0.4606.81

! Doctor found issues in 1 category.
```
</details>

https://user-images.githubusercontent.com/4605213/137899370-9bcc623f-d324-4f54-a696-0b754554e469.mov

### PR DESCRIPTION
**Pull request by yehorh**
This PR adds to the Draggable new parameter, the feedbackCursor, which changes a cursor when a pointer over the feedback widget. This is necessary because the feedback becomes invisible to the hit test, which means that the cursor cannot be redefined further down of the widget tree.

Fix #92083

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 1]
```diff
@@ -217,6 +218,13 @@ class Draggable<T extends Object> extends StatefulWidget {
   /// The data that will be dropped by this draggable.
   final T? data;
 
+  /// The mouse cursor for mouse pointers that are over the [feedback].
+  ///
+  /// When a mouse is over the [feedback], its cursor will be changed to the [cursor].
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/drag_target.dart:223]: This seems to be redundant with the first sentence?

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 2]
```diff
@@ -217,6 +218,13 @@ class Draggable<T extends Object> extends StatefulWidget {
...
+  ///
+  /// The [cursor] defaults to [MouseCursor.defer]
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/drag_target.dart:225]: cursor -> feedbackCursor?

**Comment by goderbauer** [comment]
Looks like the docs check is also unhappy, can you take a look please?

File: packages/flutter/lib/src/widgets/drag_target.dart (hunk 2 above)
Comment (yehorh) [reply packages/flutter/lib/src/widgets/drag_target.dart]: > cursor -> feedbackCursor?

Fixed 😨

File: packages/flutter/lib/src/widgets/drag_target.dart (hunk 1 above)
Comment (yehorh) [reply packages/flutter/lib/src/widgets/drag_target.dart]: > This seems to be redundant with the first sentence?

Rewrote, I hope it will be cleaner.

**Comment by yehorh** [comment]
> Looks like the docs check is also unhappy, can you take a look please?

Now it should be better. I hope.

File: packages/flutter/test/widgets/draggable_test.dart [hunk 3]
```diff
@@ -2683,6 +2684,44 @@ void main() {
     await _testChildAnchorFeedbackPosition(tester: tester, left: 100.0, top: 100.0);
   });
 
+  testWidgets('Drag feedback change cursor correctly', (WidgetTester tester) async {
```
Comment (stevendz) [inline packages/flutter/test/widgets/draggable_test.dart:2687]: Just a small one, but I would say that there is an `s` missing. 
`changes` should be the correct spelling 🤔

Comment (yehorh) [reply packages/flutter/test/widgets/draggable_test.dart]: Thanks. I will be more careful next time. 😓

File: packages/flutter/lib/src/widgets/drag_target.dart [hunk 4]
```diff
@@ -217,6 +218,11 @@ class Draggable<T extends Object> extends StatefulWidget {
   /// The data that will be dropped by this draggable.
   final T? data;
 
+  /// The cursor for a mouse pointer when the pointer is over the [feedback].
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/drag_target.dart:221]: Does this mean when the pointer is hovering over the `feedback`, the `child` or both? Or is it specific to when drag is active? Can you add more docs here to explain? Thanks!

Comment (yehorh) [reply packages/flutter/lib/src/widgets/drag_target.dart:221]: @Piinks 
But the `feedback` widget is visible only when drag is active 😓. How I can clearly explain this without [empty-pose](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose)?

btw
I would like to say thank you for the episode 53 on the Boring Show 🙏

**Comment by Piinks** [comment]
I see this comment in an email about this change, but not here:

> @Piinks
But the feedback widget is visible only when drag is active 😓. How I can clearly explain this without empty-pose?

Good point! I don't think it would be wasteful to clarify this is the documentation. 

Currently, it says:

> The cursor for a mouse pointer when the pointer is over the [feedback].

I think the *over* part might be what is confusing. If the feedback widget is only visible when drag is active, when would the pointer not be *over* it? Making this a little clearer and easier to understand is what I was suggesting. :)

**Comment by goderbauer** [comment]
@yehorh Do you have plans to come back to this and update the docs per @Piinks' feedback?

**Comment by goderbauer** [comment]
@Piinks as discussed in triage, can you pick this up and close it?

**Pull request closed without merging**